dts/riscv32/riscv32-fe310.dtsi

```diff
@@ -35,6 +35,10 @@
 		compatible = "SiFive,FE310G-0002-Z0-soc", "fe310-soc",
 			"sifive-soc", "simple-bus";
 		ranges;
+		flash0: flash@20400000 {
```

Comment:
What is this flash?  I thought the fe310 had an off chip qspi or something like that?

Reply:
I think this flash refers to the region of the QSPI that's reserved for the application.

Reply:
This might be better served by recognizing the memory addressed by the spi0 node as the FLASH_BASE_ADDRESS and then configuring 0x400000 as the FLASH_LOAD_OFFSET. Which leads me back to my question of whether we should use CONFIG_FLASH_BASE_ADDRESS on RISC-V like other arches.

Reply:
This is still not correct, look at `boards/arm/mimxrt1060_evk/mimxrt1060_evk.dts`.  

Take a look at:
```
&flexspi0 {
        reg = <0x402a8000 0x4000>, <0x60000000 0x800000>;
        is25wp064: is25wp064@0 {
                compatible = "issi,is25wp064", "jedec,spi-nor";
                reg = <0>;
                status = "ok";
        };
};
```

Reply:
You are right, we were mistaken.
FE-310 does not have a built-in flash, it's a part of HiFive board.

I'm thinking of splitting the dts into two files: 

- one for the SoC itself (without flash) 
- and second for the board (it would include SoC's dts and add flash).

Is this what you meant by referencing `mimxrt1060_evk.dts`?

Reply:
Since there had already been a dts for HiFive1 board, I just moved flash definition there and added it under `spi0` (just like in the `flexspi0` example).

Reply:
I had to duplicate the flash definiton in `qemu_riscv32.dts` as it is based on `riscv32-fe310.dtsi`.

Reply:
@galak Is this solution what you expected?